### PR TITLE
Fix getting cgroup pids

### PR DESCRIPTION
**Pull request by timstclair**
Fixes https://github.com/kubernetes/kubernetes/issues/35214, https://github.com/kubernetes/kubernetes/issues/33232

Verified manually, but I didn't have time to run all the e2e's yet (will check it in the morning).

This should be cherry-picked into 1.4, and merged into 1.5 (/cc @saad-ali )

```release-note
Fix fetching pids running in a cgroup, which caused problems with OOM score adjustments & setting the /system cgroup ("misc" in the summary API).
```

/cc @kubernetes/sig-node

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36551)

### CODE REVIEW

**Comment by saad-ali** [comment]
> This should be cherry-picked into 1.4, and merged into 1.5 (/cc @saad-ali )

Ack. Bug fix ok for post-code freeze merge. 

CC @jessfraz for 1.4 cherry pick

**Comment by jessfraz** [comment]
awesome!!

File: pkg/kubelet/cm/container_manager_linux.go [hunk 1]
```diff
@@ -443,16 +444,11 @@ func (cm *containerManagerImpl) setupNode() error {
...
-				Parent: "/",
-				Name:   "/",
-			},
-		}
 		cont.ensureStateFunc = func(manager *fs.Manager) error {
-			return ensureSystemCgroups(rootContainer, manager)
+			return ensureSystemCgroups("/", manager)
 		}
 		systemContainers = append(systemContainers, cont)
+	} else {
```
Comment (feiskyer) [inline pkg/kubelet/cm/container_manager_linux.go]: remove this else block?

File: pkg/util/cgroups/cgroups_linux.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// Forked from opencontainers/runc/libcontainer/cgroup/fs.Manager.GetPids()
+func GetPids(cgroupPath string) ([]int, error) {
+	dir, err := getCgroupPath(cgroupPath)
+	if err != nil {
+		return nil, err
+	}
+	return libcontainercgroups.GetPids(dir)
+}
+
+func getCgroupPath(cgroupPath string) (string, error) {
```
Comment (feiskyer) [inline pkg/util/cgroups/cgroups_linux.go]: nit: add documents for getCgroupPath and getCgroupParentPath

File: pkg/util/cgroups/cgroups_linux.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/*
```
Comment (vishh) [inline pkg/util/cgroups/cgroups_linux.go]: nit: I'd prefer cgroups logic to be contained inside `pkg/kubelet/cm/`

File: pkg/kubelet/cm/container_manager_linux.go (hunk 1 above)
Comment (vishh) [inline pkg/kubelet/cm/container_manager_linux.go]: +1

File: pkg/util/cgroups/cgroups_linux.go (hunk 3 above)
Comment (timstclair) [inline pkg/util/cgroups/cgroups_linux.go]: Done.

File: pkg/kubelet/cm/container_manager_linux.go (hunk 1 above)
Comment (timstclair) [inline pkg/kubelet/cm/container_manager_linux.go]: Oops! Thanks, done.

File: pkg/util/cgroups/cgroups_linux.go (hunk 2 above)
Comment (timstclair) [inline pkg/util/cgroups/cgroups_linux.go]: Done.

File: pkg/util/cgroups/cgroups_linux.go (hunk 3 above)
Comment (vishh) [inline pkg/util/cgroups/cgroups_linux.go]: @timstclair will a sub-package of `cm` work? Having all cgroups logic in there will help with code structure.

Comment (timstclair) [inline pkg/util/cgroups/cgroups_linux.go]: Sure, I don't have a strong opinion on where the package is.

Comment (timstclair) [inline pkg/util/cgroups/cgroups_linux.go]: Done.

**Comment by timstclair** [comment]
Verified OOM tests pass on all images with this fix.

**Review by vishh — APPROVED** [review]

**Comment by timstclair** [comment]
Squashed commits. I kept the BUILD files in a separate commit for easier cherry-picking to 1.4. Reapplying LGTM.

**Comment by jessfraz** [comment]
thanks!

**Pull request merged**